Script/TypeScript/AtomicExtended.d.ts

```diff
@@ -0,0 +1,115 @@
+// Manual typescript definitions from various CPP and JS files that can't yet be autogenerated
+// no-default-lib and modified lib.d.ts used to allow redefinition of self
+/// <reference no-default-lib="true"/>
+/// <reference path="./lib.d.ts" />
+/// <reference path="./Atomic.d.ts" />
+
+// From JSAtomic.cpp
+declare function print(val: any);
+declare function assert();
+declare function js_module_read_file(path: string);
+declare function openConsoleWindow();
+declare function getVM(): Atomic.JSVM;
+declare function getEngine(): Atomic.Engine;
+declare function getGraphics(): Atomic.Graphics;
+declare function getRenderer(): Atomic.Renderer;
+declare function getResourceCache(): Atomic.ResourceCache;
+declare function getInput(): Atomic.Input;
+declare function getFileSystem(): Atomic.FileSystem;
+declare function getNetwork(): Atomic.Network;
+declare function script(script: string): boolean;
+declare function destroy(node: Atomic.Node): boolean;
+declare function destroy(scene: Atomic.Scene): boolean;
+declare function destroy(component: Atomic.JSComponent): boolean;
+
+// From JSRequire.cpp
+declare function require(path: string): any;
+
+declare module Atomic {
+    // From AtomicGame.js
+    export class Game {
+        engine: Engine;
+        cache: ResourceCache;
+        renderer: Renderer;
+        graphics: Graphics;
+        input: Input;
+        uiView: UIView;
+        scene: Scene;
+        cameraNode: Node;
+        camera: Camera;
+        viewport: Viewport;
+
+        constructor();
+        init(start: any, update: any);
+        getSpriteSheet2D(xmlFile: string): Resource;
+        getSpriteSheet(xmlFile: string): Resource;
+        getSound(soundFile: string): Resource;
+        getSprite2D(spriteFile: string): Resource;
+        showDebugHud();
+        createScene2D(): Scene;
+        dumpMetrics();
+        createScene3D(filename: string): Scene;
+    }
+    var game: Game;
```

Comment:
Atomic game won't be provided in core anymore, it is really just some convenience stuff that people can include in there game as a module.  Atomic module itself now holds shortcuts to graphics, input, etc subsystems